### PR TITLE
[md5] Add new port

### LINKED ISSUE #28682: New Port Request:md5
Library name: md5

Library description: MD5 C++ class convert from Frank Thilo (thilo@unix-ag.org) for bzflag (http://www.bzflag.org)

Source repository URL: https://github.com/yaoyao-cn/md5

Project homepage (if different from the source repository): https://www.bzflag.org/

the library uses modern cmake and is easy to add to port, I have test this library on windows , linux and macOS platform

this is md5 port config file I use now: https://github.com/yaoyao-cn/vcpkg/tree/master/ports/md5

### PR DESCRIPTION
**Pull request by zhaoy-dev**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #28682

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by yaoyao-cn** [comment]
> @yaoyao-cn please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: ports/md5/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "md5",
```
Comment (LilyWangLL) [inline ports/md5/vcpkg.json:2]:
```suggestion
  "name": "yaoyao-cn-md5",
```

Comment (LilyWangLL) [inline ports/md5/vcpkg.json:2]:
```suggestion
  "name": "yaoyao-cn-md5",
```

Comment (LilyWangLL) [reply ports/md5/vcpkg.json:2]: The name md5 is easy to conflict with other ports, I suggest changing to a combination of repo names yaoyao-cn-md5. And the include folder name and share folder name also should change to same with new name yaoyao-cn-md5, you could modify them in portfile.cmake.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/md5/vcpkg.json (hunk 1 above)
Comment (yaoyao-cn) [reply ports/md5/vcpkg.json:2]: what about simple-md5 ?

Comment (LilyWangLL) [reply ports/md5/vcpkg.json:2]: Using `repo-port` is a better choice, it makes users to better distinguish between ports.

**Comment by yaoyao-cn** [comment]
it is not easy to change the port name by just modify  portfile.cmake file. not only the include folder name and share folder name, but also the cmake target name, output lib name should be changed. if so, all my code that use md5 shoud be changed. I don't have much spare time now

**Pull request closed without merging**